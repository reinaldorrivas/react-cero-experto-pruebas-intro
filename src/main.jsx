import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";

import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <FirstApp
      title="¡Hola a todos!"
      subTitle="¡Toca estudiar!"
      name="René Rivas"
    />
  </React.StrictMode>
);
