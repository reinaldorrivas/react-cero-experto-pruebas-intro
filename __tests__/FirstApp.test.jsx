import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Prueba Unitaria - FirstApp:", () => {
  test("Debe coincidir con la captura (Snapshot):", () => {
    const title = "¡Hola a todos!";
    const subTitle = "¡Toca estudiar!";
    const name = "René Rivas";

    const { container } = render(
      <FirstApp title={title} subTitle={subTitle} name={name} />
    );

    expect(container).toMatchSnapshot();
  });

  test("Debe de existir el texto para el título:", () => {
    const title = "¡Hola a todos!";
    const subTitle = "¡Toca estudiar!";
    const name = "René Rivas";

    const { getByText } = render(
      <FirstApp title={title} subTitle={subTitle} name={name} />
    );

    expect(getByText(title)).toBeTruthy();
  });

  test("Debe mostrar el título en un <h1></h1>:", () => {
    const title = "¡Hola a todos!";
    const subTitle = "¡Toca estudiar!";
    const name = "René Rivas";

    const { container } = render(
      <FirstApp title={title} subTitle={subTitle} name={name} />
    );

    const h1 = container.querySelector("h1");

    expect(h1.innerHTML).toContain(title);
  });
});
