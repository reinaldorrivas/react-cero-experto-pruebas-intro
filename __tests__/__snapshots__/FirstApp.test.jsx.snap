// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Unitaria - FirstApp: Debe coincidir con la captura (Snapshot): 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    ¡Hola a todos!
  </h1>
  <p>
    ¡Toca estudiar!
  </p>
  <p>
    René Rivas
  </p>
</div>
`;
